### PR TITLE
Task/feat get user neighbors method

### PR DESCRIPTION
**Pull request by nebHailemariam**
Hello @zhenghaoz,

I've noticed Gorse.NET is missing a method for retrieving similar users, like the API endpoint api/user/{user}/neighbors?n=10. To address this, I've submitted a pull request to add this feature and help make the library more complete.

Additionally, I've renamed the class UserScore to Neighbor instead of Score, as suggested earlier on the previous pull request, to avoid confusion with the Score field in the class. I’ve also refactored the code for better readability by adding proper spacing between methods.

Looking forward to your feedback!

### CODE REVIEW

File: Gorse.NET/Gorse.cs [hunk 1]
```diff
@@ -21,6 +21,12 @@ public override string ToString()
     }
 }
 
+public class Neighbor
```
Comment (zhenghaoz) [inline Gorse.NET/Gorse.cs:24]: Let us use `Score` as class name.

**Review by zhenghaoz — CHANGES_REQUESTED** [review]

File: Gorse.NET/Gorse.cs (hunk 1 above)
Comment (nebHailemariam) [reply Gorse.NET/Gorse.cs:24]: public class Score
{
    public string UserId { get; set; }
    public double Score { get; set; }
}
C# doesn't allow members (properties, methods, etc.) to have the same name as the class.

Comment (nebHailemariam) [reply Gorse.NET/Gorse.cs:24]: Should I use a JSON Property as follows

public class Score
{
public string UserId { get; set; }
[JsonProperty("Score")]
public double UserScore { get; set; }
}

Comment (nebHailemariam) [reply Gorse.NET/Gorse.cs]: No worries, I've changed it back to UserScore!

**Review by zhenghaoz — APPROVED** [review]

**Pull request merged**